datasets/code_x_glue_cc_clone_detection_poj_104/code_x_glue_cc_clone_detection_poj_104.py

```diff
@@ -0,0 +1,110 @@
+import json
+import os
+import os.path
+from typing import List
+
+import datasets
+
+from .common import TrainValidTestChild
+from .generated_definitions import DEFINITIONS
+
+
+_DESCRIPTION = """Given a code and a collection of candidates as the input, the task is to return Top K codes with the same semantic. Models are evaluated by MAP score.
+We use POJ-104 dataset on this task."""
+
+_CITATION = """@inproceedings{mou2016convolutional,
+title={Convolutional neural networks over tree structures for programming language processing},
+author={Mou, Lili and Li, Ge and Zhang, Lu and Wang, Tao and Jin, Zhi},
+booktitle={Proceedings of the Thirtieth AAAI Conference on Artificial Intelligence},
+pages={1287--1293},
+year={2016}
+}"""
+
+
+class CodeXGlueCcCloneDetectionPoj104(TrainValidTestChild):
+    _DESCRIPTION = _DESCRIPTION
+    _CITATION = _CITATION
+
+    _FEATURES = {
+        "id": datasets.Value("int32"),  # Index of the sample
+        "code": datasets.Value("string"),  # The full text of the function
+        "label": datasets.Value("string"),  # The id of problem that the source code solves
+    }
+
+    _SUPERVISED_KEYS = ["label"]
+
+    SPLIT_RANGES = {"train": (1, 65), "valid": (65, 81), "test": (81, 195)}
+
+    def generate_urls(self, split_name):
+        yield "data", "programs.tar.gz"
+
+    def pregenerate_all(self, root_path):
+        def files(path):
+            g = os.walk(path)
+            file = []
+            for path, dir_list, file_list in g:
+                for file_name in file_list:
+                    file.append(os.path.join(path, file_name))
+            return file
+
+        cont = 0
+        for split_name, range_info in self.SPLIT_RANGES.items():
+            with open(os.path.join(root_path, f"{split_name}.jsonl"), "w") as f:
+                for i in range(*range_info):
+                    items = files(os.path.join(root_path, "ProgramData/{}".format(i)))
+                    for item in items:
+                        js = {}
+                        js["label"] = item.split("/")[1]
+                        js["index"] = str(cont)
+                        js["code"] = open(item, encoding="latin-1").read()
+                        f.write(json.dumps(js) + "\n")
+                        cont += 1
```

Comment:
It would be nice to not have to write new files.  This loop should be used to `yield` example instead of writing to a jsonl file

Reply:
What about if this was all moved to the `_split_generators` function since this is essentially trying to do some data wrangling to get it into an easier to use format?

Reply:
I'd move it to the `_generate_examples` method actually. It is the one that yields example